src/components/listitems/ListItemSwitch.tsx

```diff
@@ -10,14 +15,25 @@ import { IOIcons, Icon } from "../icons";
 import { HSpacer, VSpacer } from "../spacer";
 import { H6, LabelSmall, LabelLink } from "../typography";
 import { NativeSwitch } from "../switch/NativeSwitch";
+import { Badge } from "../badge";
 
 type Props = {
   label: string;
   onSwitchValueChange?: (newValue: boolean) => void;
   description?: string;
   icon?: IOIcons;
   action?: SwitchAction;
-};
+  isLoading?: boolean;
+} & (
+  | {
+      badgeText: string;
+      badgeVariant: Pick<Badge, "variant">["variant"];
+    }
+  | {
+      badgeText?: undefined;
+      badgeVariant?: undefined;
```

Comment:
```suggestion
      badgeText?: never;
      badgeVariant?: never;
```

what about using type `never` in this case?

Reply:
That's absolutely better! Addressed in https://github.com/pagopa/io-app-design-system/pull/79/commits/35235694105bde937655ca7f0c2e3e2214d98813